src/myprofile-page/CreateMutationHelper.js

```diff
@@ -0,0 +1,115 @@
+import { gql } from '@apollo/client';
+import Moment from 'moment';
+import { PLACEHOLDER_MUTATION } from '../sources';
+
+export const createUserEducations = (userId, educations) => {
+  if (!(educations && educations.length > 0)) {
+    return PLACEHOLDER_MUTATION;
+  }
+  const mutations = [];
+  for (let index = 0; index < educations.length; index += 1) {
+    const {
+      institution, description, startDate, endDate,
+    } = educations[index];
+    const mutation = `
+                education${index}: createUserEducation(
+                        userId: ${userId},
+                        institution: "${institution}",
+                        description: "${description}",
+                        startDate: "${Moment(new Date(startDate) * 1000).format('YYYY-MM-DD hh:mm:ss')}",
+                        endDate: "${Moment(new Date(endDate) * 1000).format('YYYY-MM-DD hh:mm:ss')}",
+                    ){
+                    id
+                },
+            `;
+    mutations.push(mutation);
+  }
+
+  return gql`
+        mutation createUserEducation{
+            ${mutations}
+        }
+    `;
+};
+
+export const createUserExperiences = (userId, experiences) => {
+  if (!(experiences && experiences.length > 0)) {
+    return PLACEHOLDER_MUTATION;
+  }
+  const mutations = [];
+  for (let index = 0; index < experiences.length; index += 1) {
+    const {
+      institution, description, startDate, endDate,
+    } = experiences[index];
+    const mutation = `
+                experience${index}: createUserWorkExperience(
+                        userId: ${userId},
+                        institution: "${institution}",
+                        description: "${description}",
+                        startDate: "${Moment(new Date(startDate) * 1000).format('YYYY-MM-DD hh:mm:ss')}",
+                        endDate: "${Moment(new Date(endDate) * 1000).format('YYYY-MM-DD hh:mm:ss')}",
+                    ){
+                    id
+                },
+            `;
+    mutations.push(mutation);
+  }
+
+  return gql`
+        mutation createUserWorkExperiences{
+            ${mutations}
+        }
+    `;
+};
+
+export const createSkills = (userId, skills) => {
+  if (!(skills && skills.length > 0)) {
+    return PLACEHOLDER_MUTATION;
+  }
+
+  const mutations = [];
+
+  for (let index = 0; index < skills.length; index += 1) {
+    const {
+      skill,
+    } = skills[index];
+    const mutation = `
+                skill${index}: createSkill(
+                        name: "${skill.name}"
+                    ){
+                    id
+                },
+            `;
+    mutations.push(mutation);
+  }
+
+  return gql`
+    mutation createSkills{
+      ${mutations}
+    }
+  `;
+};
+
+export const createUserSkills = (userId, newSkills, userSkills) => {
```

Comment:
I would call this something like "alignSkillsToUser"

Reply:
updated, was a little confusing that these two functions had similar names too.